tests/rptest/tests/offset_retention_test.py

```diff
@@ -181,3 +183,77 @@ def offsets_exist():
         # after waiting for twice the retention period, it should be gone
         time.sleep(self.period * 2 - self.period / 2)
         assert not offsets_exist()
+
+
+class OffsetDeletionTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=3), )
+    group = "hey_hey_group"
+
+    def __init__(self, test_context):
+        super(OffsetDeletionTest, self).__init__(test_context=test_context)
+
+        self.rpk = RpkTool(self.redpanda)
+        self.kcl = KCL(self.redpanda)
+
+    @cluster(num_nodes=4)
+    def test_offset_deletion(self):
+        def offsets_exist():
+            desc = self.rpk.group_describe(self.group)
+            return len(desc.partitions) > 0
+
+        # Produce some data to a topic and consume assigning new consumer group
+        for _ in range(0, 10):
+            self.rpk.produce(self.topic, "k", "v", partition=0)
+            self.rpk.produce(self.topic, "k", "v", partition=1)
+            self.rpk.produce(self.topic, "k", "v", partition=2)
+        self.rpk.consume(self.topic, n=3, group=self.group)
+        wait_until(offsets_exist, timeout_sec=1, backoff_sec=1)
+
+        # Assert offset-delete errors when request contains missing topic-partitions
+        missing_topic = f"{self.topic}-foo"
+        missing_topic_partitions = {missing_topic: [0, 1, 2]}
+        output = self.kcl.offset_delete(self.group, missing_topic_partitions)
+        assert len(output) == 3
+        self.redpanda.logger.info("O: %s" % output)
+        for response in output:
+            assert response.topic == missing_topic, f"Expected: {missing_topic} Observed: {response.topic}"
+            assert response.status == 'UNKNOWN_TOPIC_OR_PARTITION', response.status
+
+        # Assert offset-delete errors when non-existent group is passed in
+        topic_partitions = {self.topic: [0, 1, 2]}
+        try:
+            self.kcl.offset_delete("missing", topic_partitions)
+            assert False, "Expected KCL exception"
+        except KCLException as e:
+            assert e.error == 'GROUP_ID_NOT_FOUND', e.error
+
+        # Assert offset-delete errors when attempting to delete offsets of topic
+        # partitions still assigned to an active group
+        consumer = KafkaCliConsumer(
+            self.test_context,
+            self.redpanda,
+            topic=self.topic,
+            group=self.group,
+            from_beginning=True,
+            instance_name='cli-consumer-offset-delete-test')
+        consumer.start()
+        consumer.wait_for_messages(1)
+
+        output = self.kcl.offset_delete(self.group, topic_partitions)
+        assert len(output) == 3
+        for response in output:
+            assert response.topic == self.topic, f"Expected: {self.topic} Observed: {response.topic}"
+            assert response.status == 'GROUP_SUBSCRIBED_TO_TOPIC', response.status
+
+        consumer.stop()
+        consumer.wait()
+
+        # Assert offset-delete removes offsets of dead groups
```

Comment:
Add another test case testing removal of offsets of alive groups that have unsubscribed to the desired offsets to be deleted